templates/todo/api/csharp-cosmos-sql/TodoEndPointsExt.cs

```diff
@@ -0,0 +1,164 @@
+using SimpleTodo.Api;
+
+namespace Todo.Api
+{
+    public static class TodoEndpointsExt
+    {
+        public static RouteGroupBuilder MapTodosApi(this RouteGroupBuilder group)
+        {
+            group.MapGet("/", GetLists);
+            group.MapPost("/", CreateList);
+            group.MapGet("/{list_id}", GetList);
+            group.MapPut("/{list_id}", UpdateList);
+            group.MapDelete("/{list_id}", DeleteList);
+            group.MapGet("/{list_id}/items", GetListItems);
+            group.MapPost("/{list_id}/items", CreateListItem);
+            group.MapGet("/{list_id}/items/{item_id}", GetListItem);
+            group.MapPut("/{list_id}/items/{item_id}", UpdateListItem);
+            group.MapDelete("/{list_id}/items/{item_id}", DeleteListItem);
+            group.MapGet("/{list_id}/state/{state}", GetListItemsByState);
+            return group;
+        }
+        public static async Task<Ok<IEnumerable<TodoList>>> GetLists(ListsRepository _repository, int? skip = null, int? batchSize = null)
+        {
+            return TypedResults.Ok(await _repository.GetListsAsync(skip, batchSize));
+        }
+
+        public static async Task<IResult> CreateList(ListsRepository _repository, CreateUpdateTodoList list)
+        {
+            var todoList = new TodoList(list.name)
+            {
+                Description = list.description
+            };
+
+            await _repository.AddListAsync(todoList);
+
+            return TypedResults.Created($"/lists/{todoList.Id}", todoList);
+        }
+
+        public static async Task<IResult> GetList(ListsRepository _repository, string list_id)
+        {
+            var list = await _repository.GetListAsync(list_id);
+
+            return list == null ? TypedResults.NotFound() : TypedResults.Ok(list);
+        }
+
+        public static async Task<IResult> UpdateList(ListsRepository _repository, string list_id, CreateUpdateTodoList list)
+        {
+            var existingList = await _repository.GetListAsync(list_id);
+            if (existingList == null)
+            {
+                return TypedResults.NotFound();
+            }
+
+            existingList.Name = list.name;
+            existingList.Description = list.description;
+            existingList.UpdatedDate = DateTimeOffset.UtcNow;
+
+            await _repository.UpdateList(existingList);
+
+            return TypedResults.Ok(existingList);
+        }
+
+        public static async Task<IResult> DeleteList(ListsRepository _repository, string list_id)
+        {
+            if (await _repository.GetListAsync(list_id) == null)
+            {
+                return TypedResults.NotFound();
+            }
+
+            await _repository.DeleteListAsync(list_id);
+
+            return TypedResults.NoContent();
+        }
+
+        public static async Task<IResult> GetListItems(ListsRepository _repository, string list_id, int? skip = null, int? batchSize = null)

```

Comment:
Can we change the _ convention, it's not C# like to use _ in method paramters. Also the naming `list_id` is not idiomatic either. These should be:

```suggestion
        public static async Task<IResult> GetListItems(ListsRepository repository, string listId, int? skip = null, int? batchSize = null)

```

Reply:
Sure, I just followed the convention the sample already used